sim/main_game.py

```diff
@@ -5,49 +5,36 @@
 from characters.player.player import Player
 from characters.npc.npc import Npc
 
-INPUTBOX_DISPLAY = False
-
 
 class projectG:
-    """
-    Class to manage game assets and behavior
-    """
-
     def __init__(self):
-        """
-        Initialize game and create game resources
-        """
         pygame.init()
-
         self.settings = GameSettings()
         self.input_box_settings = InputBox()
-
         self.screen = pygame.display.set_mode(
             (self.settings.screen_width, self.settings.screen_height)
         )
         pygame.display.set_caption("ProjectG")
         self.player = Player(self)
-        self.type_box = InputBox(self)
+        self.type_box = InputBox()
         self.npc = Npc(self)
+        self.collided = False
+        self.font = pygame.font.Font(None, 32)
+        self.input_box = pygame.Rect(100, 100, 140, 32)
+        self.text = "need to be able to update"
```

Comment:
did not manage to make the text box editable yet, can you check?

Reply:
> did not manage to make the text box editable yet, can you check?

Works now:

<img width="1312" alt="image" src="https://github.com/abaddonpuff/project_G/assets/387927/8aaea526-6055-43b0-81f5-5499984dc677">

Reply:
then move away, it hides
then move to collision again, it shows up with history
todo: database persistence for later